docs(llma): surveys integration (#14886)
PostHog/posthog.com: contents/docs/llm-analytics/collect-user-feedback.mdx

---
title: Collect user feedback
---

import CalloutBox from 'components/Docs/CalloutBox'
import EmbeddedSurvey from 'components/Docs/EmbeddedSurvey'

<CalloutBox icon="IconInfo" title="User feedback is in beta" type="fyi">

User feedback is currently in beta. We'd love to hear your feedback as we develop this feature:

> <EmbeddedSurvey surveyId="019c4416-db5e-0000-4b3c-3a565a512fed" />

Author: ha yeah we need to set some better defaults on the iframe'd surveys haha

</CalloutBox>

> To collect user feedback for LLM traces, PostHog provides a native integration with [Surveys](/docs/surveys) that enables you to attach user feedback, including thumbs up/down responses, directly to a given trace.

Reviewer: ```suggestion
To collect user feedback for LLM traces, PostHog provides a native integration with [Surveys](/docs/surveys). The Surveys integration enables you to collect user feedback and thumbs up/down responses attached directly to a given trace.
```

readability.

## How it works

1. Create a survey for a group of LLM traces.
2. Send survey response events with an `$ai_trace_id` property.
3. View user feedback per-trace within LLM Analytics, or view all feedback within [Surveys](https://app.posthog.com/surveys).

> For the common thumbs up/down use-case, we provide a React hook `useThumbSurvey` to quickly attach a thumb survey and optionall followup questions to your existing UI. In other cases, you can manually send survey-related events to attach them to your trace.

Reviewer: ```suggestion
For the common thumbs up/down use-case, we provide a React hook `useThumbSurvey` to quickly attach a thumb survey and optional followup questions to your existing UI. In other cases, you can manually send survey-related events to attach them to your trace.
```
